vendor/bat-native-ledger/src/bat/ledger/internal/endpoints/gemini/post_commit_transaction/post_commit_transaction_gemini.cc

```diff
@@ -0,0 +1,93 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/endpoints/gemini/post_commit_transaction/post_commit_transaction_gemini.h"
+
+#include <utility>
+
+#include "base/base64.h"
+#include "base/json/json_reader.h"
+#include "base/json/json_writer.h"
+#include "bat/ledger/internal/endpoint/gemini/gemini_utils.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "net/http/http_status_code.h"
+
+namespace ledger::endpoints {
+using Error = PostCommitTransactionGemini::Error;
+using Result = PostCommitTransactionGemini::Result;
+
+namespace {
+
+Result ParseBody(const std::string& body) {
```

Comment:
is all this JSON parsing happening in a non-main process? need to follow https://chromium.googlesource.com/chromium/src/+/master/docs/security/rule-of-2.md

Reply:
Hmm, the parsing happens in the utility process (`bat_ledger.mojom.BatLedgerService`), which is launched as `--service-sandbox-type=none` (`sandbox::mojom::Sandbox::kNoSandbox`) for non-Android (and as `--service-sandbox-type=utility` (`sandbox::mojom::Sandbox::kUtility`) for Android), which I suspect is a problem then, as `--service-sandbox-type=none` is pretty much the same rights as the browser itself?

Reply:
yeah i think this is a problem... do we know if it needs sandbox disabled for some reason? another option is to use the JSON sanitizer like the wallet does, but sandboxed utility process is preferable

Reply:
For some reason, I completely forgot that @kdenhartog has already pointed this out [here](https://github.com/brave/brave-core/pull/15101#discussion_r979105354), which I already addressed in [this](https://github.com/brave/brave-core/pull/15261) PR.